kex-runner/src/main/kotlin/org/vorpal/research/kex/jacoco/minimization/GreedyTestReductionImpl.kt

```diff
@@ -39,14 +41,17 @@ class GreedyTestReductionImpl : TestSuiteMinimizer {
 
         var satisfiedReq = 0
         val importantTests = mutableSetOf<Path>()
-        while (satisfiedReq < requestSet.size) {
+        val maxTests = kexConfig.getIntValue("testGen", "maxTests", Integer.MAX_VALUE)
+        while (satisfiedReq < requestSet.size && importantTests.size < maxTests) {
             val (maxTestPath, maxTest) = tests.maxByOrNull { it.value.power } ?: break
             if (maxTest.power == 0) break
 
             satisfiedReq += maxTest.power
             maxTest.reqs.forEach { requirements[it]!!.visit(tests) }
             importantTests.add(maxTestPath)
         }
+        // TODO don't use fragile string comparisons
```

Comment:
It is better to use something like [this](https://github.com/vorpal-research/kex/blob/8a9704c9a66f0239a72eb41fb0f74ca01e5b0a57/kex-runner/src/main/kotlin/org/vorpal/research/kex/jacoco/minimization/TestSuiteMinimizer.kt#L24) to ignore fragile string comparisons

Reply:
Actually, I think that there is not need to explicitly check these classes here, because they will not be deleted by `TestSuiteMinimizer.deleteTestCases(reducedTests)`. Previously there was a bug in there that did not consider `EqualityUtils`, but now everything should work fine.

Reply:
Removed the change in `GreedyTestReductionImpl`, and fixed another bug in `TestSuiteMinimizer`